lib/smart_proxy_pulp_plugin/disk_usage.rb

```diff
@@ -0,0 +1,33 @@
+module PulpProxy
+  class DiskUsage
+    include ::Proxy::Util
+    SIZE = {:kilobyte => 1_024, :megabyte => 1_048_576, :gigabyte => 1_073_741_824, :terabyte => 1_099_511_627_776}
+
+    attr_reader :path, :stat, :size
+
+    def initialize(opts ={})
+      @path = opts[:path] || PulpProxy::Plugin.settings.pulp_dir
+      @size = opts[:size] || SIZE[:megabyte]
+      get_stat
+    end
+
+    def to_json
+      data.to_json
+    end
+
+    def data
+      stat.merge({:path => path})
+    end
+
+    private
+
+
+    def get_stat
+      command_path = which("df")
+      raw = `#{command_path} -B#{size} #{path}`.split("\n")
```

Comment:
you should verify which size is acceptable, also, this opens the API to a vounrability, as a user can send path = ```/path & some``` other command